src/styles/blocks/main.scss

```diff
@@ -0,0 +1,32 @@
+.catalog {
+  padding: 50px 40px;
+  display: grid;
+  grid-template-columns: repeat(auto-fill, 200px);
+  gap: 46px 48px;
+  justify-content: center;
+  justify-items: center;
+}
+
+@media (max-width: 487px) {
+  .catalog {
+    grid-template-columns: repeat(1, 200px);
+  }
+}
+
+@media (min-width: 488px) {
+  .catalog {
+    grid-template-columns: repeat(2, 200px);
+  }
+}
+
+@media (min-width: 768px) {
+  .catalog {
+    grid-template-columns: repeat(3, 200px);
+  }
+}
+
+@media (min-width: 1024px) {
+  .catalog {
+    grid-template-columns: repeat(4, 200px);
+  }
+}
```

Comment:
The media query at line 28 is correct, but it could be optimized. Instead of setting the grid-template-columns property to repeat(4, 200px), you could set it to repeat(auto-fit, minmax(200px, 1fr)). This will create as many 200px columns as possible, but will also expand those columns to fill any remaining space if there aren't enough items to fill them.